app/controllers/api/avatars.php

```diff
@@ -48,9 +48,10 @@
 
     $cache = new Cache(new Filesystem(APP_STORAGE_CACHE . '/app-0')); // Limit file number or size
     $data = $cache->load($key, 60 * 60 * 24 * 30 * 3/* 3 months */);
-
     if ($data) {
         //$output = (empty($output)) ? $type : $output;
+        App::setResource('cacheKey', fn () => $key);
+        App::setResource('cachePath', fn () => 'app-0');
```

Comment:
This is very uncommon to how we handle resources. We usually don't manipulate them from within the action. For simple primitive strings, it might work better to use labels with special params patterns, very similar to how we use labels for abuse. We can use the cache shutdown hook to parse the pattern and populate the proper cache keys based on the request object.

Reply:
solution refactored